### PR TITLE
[script.service.tvheadend-route-latencies] 1.0.0

### PR DESCRIPTION
**Pull request by iwis**
### Description
Add-on checks quality of network connection between Kodi and Tvheadend by pinging some important routers and servers that are between them.

### Checklist:

- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-scripts/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [script.foo.bar] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practise but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-scripts/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: script.service.tvheadend-route-latencies/addon.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="script.service.tvheadend-route-latencies"
+       name="Tvheadend route latencies"
+       version="1.0.0"
+       provider-name="iwis">
+    <requires>
+        <import addon="xbmc.python" version="2.20.0" />
```
Comment (romanvm) [inline script.service.tvheadend-route-latencies/addon.xml]: It is v `2.25.0` for Krypton

File: script.service.tvheadend-route-latencies/default.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    total_requests = 1 + sec/wait_sec              #number of requests that will be sent
+    total_answers  = 0
+    packet_loss    = 0
+    total_time     = 0
+    s = requests.Session()
+    try:
+        s.get(url, timeout=5).text                 #make a TCP connection to speed up proper requests
+    except requests.RequestException:
+        pass
+    time.sleep(1)                                  #be nice
```
Comment (romanvm) [inline script.service.tvheadend-route-latencies/default.py]: Please use `xbmc.sleep()` here and in other cases

File: script.service.tvheadend-route-latencies/default.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        control.setAnimations([('WindowOpen',  'effect=fade start=0   end=100 time=300',),
+                               ('WindowClose', 'effect=fade start=100 end=0   time=300',)])
+
+    def __init__(self):
+        super(TVNetworkHealthWindow, self).__init__("Tvheadend route latencies")
+        try:  #comment it temporary to see detailed log in case of errors
+            self.testlength = int(ADDON.getSetting("testlength"))  #how long a single test will take
+
+            #--------------- IP addresses of consecutive routers and servers ---------------
+            local_router    = xbmc.getInfoLabel('Network.GatewayAddress')
```
Comment (romanvm) [inline script.service.tvheadend-route-latencies/default.py]: For some network configurations this infolabel may return incorrect data.

Comment (iwis) [inline script.service.tvheadend-route-latencies/default.py]: I noticed that on Android - xbmc.getInfoLabel('Network.GatewayAddress') returns empty string and System information > Network > Gateway is also empty in Kodi GUI. I described it in README.md. Is there another way I can get network gateway on Android?

Comment (iwis) [inline script.service.tvheadend-route-latencies/default.py]: I checked in https://github.com/xbmc/xbmc/blob/master/xbmc/network/linux/NetworkLinux.cpp that Kodi reads gateway address directly from /proc/net/route file. Unfortunately on Android this file does not contain a line for gateway now. This is why we get no info. Gateway can be read using https://pypi.python.org/pypi/netifaces/ library, but I don't know if there is possibility to use it in Kodi.

File: script.service.tvheadend-route-latencies/addon.xml (hunk 1 above)
Comment (iwis) [inline script.service.tvheadend-route-latencies/addon.xml]: Fixed and tested.

File: script.service.tvheadend-route-latencies/default.py (hunk 2 above)
Comment (iwis) [inline script.service.tvheadend-route-latencies/default.py]: Changed and tested.

File: script.service.tvheadend-route-latencies/default.py (hunk 3 above)
Comment (romanvm) [inline script.service.tvheadend-route-latencies/default.py]: Platform and version specific wheels for `netifaces` suggest that it has binary components so it is not acceptable for a Kodi addon. But `Network.GatewayAddress` limitation is not exactly a show stopper, and the rest looks good.

**Pull request merged**